Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/lucene/impl/LuceneUtility.java

```diff
@@ -537,14 +530,80 @@ public void indexConcepts(String userName)
 
         catch (IOException ex) {
             throw new LuceneException("Issues in querying lucene index. Please retry", ex);
-        } catch (ParseException e) {
-            throw new LuceneException("Issues in framing the query", e);
         }
         logger.debug("Number of concepts retrieved from lucene = " + concepts.size());
         return concepts.toArray(new ConceptEntry[concepts.size()]);
 
     }
 
+    private void buildQuery(BooleanClause.Occur occur, PerFieldAnalyzerWrapper perFieldAnalyzerWrapper,
+            QueryBuilder qBuild, BooleanQuery.Builder builder, LuceneField luceneFieldAnnotation, String searchString) {
+
+        BooleanQuery.Builder tokenizedQueryBuilder = new BooleanQuery.Builder();
+        buildTokenizedOrWildCardQuery(luceneFieldAnnotation, searchString, tokenizedQueryBuilder);
+
+        if (luceneFieldAnnotation.isTokenized() && luceneFieldAnnotation.isShortPhraseSearchable()) {
+            BooleanQuery.Builder rootQueryBuilder = new BooleanQuery.Builder();
+            rootQueryBuilder.add(tokenizedQueryBuilder.build(), Occur.SHOULD);
+
+            // Short word searching
+            BooleanQuery.Builder shortWordSearchQueryBuilder = new BooleanQuery.Builder();
+            shortWordSearchQueryBuilder
+                    .add(new PhraseQuery(luceneFieldAnnotation.lucenefieldName() + LuceneFieldNames.UNTOKENIZED_SUFFIX,
+                            searchString), Occur.SHOULD);
+
+            rootQueryBuilder.add(shortWordSearchQueryBuilder.build(), Occur.SHOULD);
+            // Adding both tokenized and untokenized queries to main query
+            // builder.
+            builder.add(rootQueryBuilder.build(), occur);
+        }
+
+        else if (luceneFieldAnnotation.isTokenized()) {
+            // Tokenized and not short word searchable
+            builder.add(tokenizedQueryBuilder.build(), occur);
+        }
+
+        else {
+            if (luceneFieldAnnotation.isWildCardSearchEnabled()) {
+                createWildCardSearchQuery(luceneFieldAnnotation, searchString, builder, occur);
+            } else {
+                builder.add(new BooleanClause(
+                        new TermQuery(new Term(luceneFieldAnnotation.lucenefieldName(), searchString)), occur));
+            }
+        }
```

Comment:
loose the 'else if' block and instead do a:
```
if is tokenized:
   build tokenized query (move that into the if statement, since you don't need it in the else block)
   if short term:
      build OR phrase
      assign root builder to tokenized builder
   end if short term
   add tokenized builder to builder
else:
  etc.
```
does this make sense?